UXToolsGame/Plugins/UXTools/Source/UXTools/Private/Interactions/UxtGenericManipulatorComponent.cpp

```diff
@@ -48,8 +48,15 @@ void UUxtGenericManipulatorComponent::TickComponent(float DeltaTime, ELevelTick
 FQuat UUxtGenericManipulatorComponent::GetViewInvariantRotation() const
 {
 	FRotator CameraSpaceYawPitchRotation = UUxtFunctionLibrary::GetHeadPose(GetWorld()).GetRotation().Rotator();
+	
+	// Previously roll was ignored but that meant that the object had it's roll offset everytime the user clicked on it/selected it
+	// when using either MaintainRotationToUser or GravityAlignedMaintainRotationToUser behaviours.
+	// The object rolled in the opposite direction to the roll on the users head.
+	// This did not produce the invariant rotation value we desired, so let's comment out the line below that resets the roll
+	// and instead use the complete world rotation value to calculate the relative orientation value.
+
 	// Ignore roll
-	CameraSpaceYawPitchRotation.Roll = 0.0f;
+	//CameraSpaceYawPitchRotation.Roll = 0.0f;
```

Comment:
I think this originated from mrtk unity - we can include roll here. for me it feels weird to restrict the roll in general, but for users that might not want this (and rely on the functionality as is) it would be good to have the option to disable roll for this calculation eg as a checkbox in the inspector of the component.

We're currently moving this logic into a constraint system similar than MRTK Unitys ObjectManipulator is using - so we can add the optional disable there as well.

Reply:
Also I would say remove the commented code and the explanation. The explanation makes more sense as a description for this PR but I wouldn't include it into the code :)